wen-restart/src/wen_restart.rs

```diff
@@ -126,27 +422,460 @@ mod tests {
+        let last_vote_bankhash = Hash::new_unique();
+        let expected_slots_to_repair: Vec<Slot> =
+            (last_vote_slot + 1..last_vote_slot + 3).collect();
+        let blockstore_clone = test_state.blockstore.clone();
+        let bank_forks_clone = test_state.bank_forks.clone();
+        let wen_restart_thread_handle = Builder::new()
+            .name("solana-wen-restart".to_string())
+            .spawn(move || {
+                assert!(wait_for_wen_restart(
+                    &wen_restart_proto_path_clone,
+                    VoteTransaction::from(Vote::new(vec![last_vote_slot], last_vote_bankhash)),
+                    blockstore_clone,
+                    cluster_info_clone,
+                    bank_forks_clone,
+                    wen_restart_repair_slots.clone(),
+                    80,
+                    Arc::new(AtomicBool::new(false)),
+                )
+                .is_ok());
+            })
+            .unwrap();
+        let mut rng = rand::thread_rng();
+        let mut expected_messages = HashMap::new();
+        // Skip the first 2 validators, because 0 is myself, we only need 8 more to reach > 80%.
+        for keypairs in test_state.validator_voting_keypairs.iter().skip(2) {
+            let node_pubkey = keypairs.node_keypair.pubkey();
+            let node = LegacyContactInfo::new_rand(&mut rng, Some(node_pubkey));
+            let last_vote_hash = Hash::new_unique();
+            let now = timestamp();
+            push_restart_last_voted_fork_slots(
+                test_state.cluster_info.clone(),
+                &node,
+                &expected_slots_to_repair,
+                &last_vote_hash,
+                &keypairs.node_keypair,
+                now,
+            );
+            expected_messages.insert(
+                node_pubkey.to_string(),
+                LastVotedForkSlotsRecord {
+                    last_voted_fork_slots: expected_slots_to_repair.clone(),
+                    last_vote_bankhash: last_vote_hash.to_string(),
+                    shred_version: SHRED_VERSION as u32,
+                    wallclock: now,
+                },
+            );
+        }
+
+        // Simulating successful repair of missing blocks.
+        let mut parent_bank = test_state.bank_forks.read().unwrap().root_bank();
+        for slot in expected_slots_to_repair {
+            let mut my_bank_forks = test_state.bank_forks.write().unwrap();
+            my_bank_forks.insert(Bank::new_from_parent(
+                parent_bank.clone(),
+                &Pubkey::default(),
+                slot,
+            ));
+            parent_bank = my_bank_forks.get(slot).unwrap();
+            parent_bank.freeze();
+        }
+
+        let _ = wen_restart_thread_handle.join();
+        let buffer = read(test_state.wen_restart_proto_path).unwrap();
         let progress = WenRestartProgress::decode(&mut std::io::Cursor::new(buffer)).unwrap();
```

Comment:
can we just call `read_wen_restart_records()` instead?

Reply:
Done.